server/server_test.go

```diff
@@ -145,11 +148,16 @@ func TestDefaultAddresses(t *testing.T) {
 }
 
 func TestErrorInstrumentationMiddleware(t *testing.T) {
+	reg := prometheus.NewRegistry()
+	prometheus.DefaultRegisterer = reg
+	prometheus.DefaultGatherer = reg
+
```

Comment:
Please explain this change somewhere.  Is it wise to change such global variables?
I don't think it is linter-driven as claimed by the commit description.